### DISCUSSION TITLE
Customize GUI (ability to move and resize all UI widgets)

### DISCUSSION THREAD

**Discussion by jordanmcleod** [PintaProject/Pinta#930] [Ideas]
I'm enjoying layers aside from the inability to change layer order (addressed by https://github.com/PintaProject/Pinta/discussions/760).  On the subject of layers: my names are often truncated by the "Layers" widget (?) in Pinta.  I've tried to resize the area but the only resizing that appears to be working is vertically and, only then, between the "Layers" and "History" sections.

I would also like to be able to move the other GUI sections that appear to be set statically.  For instance, I would love to be able to move the toolbar (select, shapes, etc.) to the right-hand side of the screen, then move the "Layers" and "History" sections to the left as well as place those two sections side-by-side and expanded to the full height of the allowed.

In general, it would be nice to be able to move and resize all UI widgets in Pinta.  The widgets would have a default "docked" location and moving them would undock (dedock?) the UI widget.  If the toolbar were resized to be less linear, the icons would change to suit like word-wrap does for text.  Widgets would be able to be moved and docked in other locations or used while not docked in a position.  Ideally there would exist an option to remember the new locations when exiting Pinta as well as the option to reset to default locations.

**Comment by cameronwhite** [comment]
It should be possible to resize the layer list horizontally, e.g. as in this screen grab. You're using Pinta 2.1, correct?

https://github.com/user-attachments/assets/190e1a15-21e5-4d76-9351-1a140d5eb3b6

**Reply by jordanmcleod** [reply]
Thank you.  I was able to resize the horizontal width, which also enabled me to learn of the feature to move layers down or up, which I had previously thought was lacking from Pinta entirely.  I am using version 2.1.1.

**Comment by cameronwhite** [comment]
Pinta used to have more of the docking functionality you described in versions 1.x, since it reused the docking library from Monodevelop. That library was very challenging to port to GTK3 for Pinta 2.0 since it had a ton of custom widgets and very complex functionality, so we ended up having to move to a simpler UI layout. Porting that library to GTK3 (and maintaining it further, e.g. with Pinta also now being ported to GTK4) is not practical with our small group of developers

The future direction is probably to add C# bindings for https://gitlab.gnome.org/GNOME/libpanel which looks promising , although I haven't done much testing of it

**Comment by Skif-off** [comment]
@cameronwhite, is it possible to just swap the "Layers" and "History" panels?

**Reply by cameronwhite** [reply]
I don't think there's a way to do that currently (again a limitation with the basic docking layout we're using currently, although probably not as hard as general moving / resizing of all panels)

**Reply by Skif-off** [reply]
Yea, I understood that. I thought maybe the idea of just swapping the panels would be interesting
(It seems to me that the layout of the panel with layers at the bottom is more common, but I'm not sure :).).
